### PR TITLE
Ginkgo: Updating vagrant to Kubernetes 1.8

### PR DESCRIPTION
**Pull request by eloycoto**
- Updating default test to kubernetes 1.8
- Added kubernetes 1.7 test on Nightly
- Some adjustments in the jenkins timeouts
- Updated cilium daemon set

Signed-off-by: Eloy Coto <eloy.coto@gmail.com>

### CODE REVIEW

**Comment by eloycoto** [comment]
test-all-ginkgo

**Comment by eloycoto** [comment]
test-me-please

File: test/provision/k8s_install.sh [hunk 1]
```diff
@@ -47,8 +48,14 @@ sudo rm /var/lib/apt/lists/lock || true
 retry_function "wget https://packages.cloud.google.com/apt/doc/apt-key.gpg"
 apt-key add apt-key.gpg
 
+sudo swapoff -a
```
Comment (ianvernon) [inline test/provision/k8s_install.sh]: Just curious, what do we need this for?

File: test/provision/k8s_install.sh [hunk 2]
```diff
@@ -47,8 +48,14 @@ sudo rm /var/lib/apt/lists/lock || true
...
+
 case $K8S_VERSION in
-    "1.6"|"1.7"|"1.8")
+    "1.6"|"1.7")
+        KUBERNETES_CNI_VERSION="0.5.1-00"
+        ;;
+    "1.8")
+        KUBEADM_OPTIONS="--skip-preflight-checks"
```
Comment (ianvernon) [inline test/provision/k8s_install.sh]: Why are preflight checks being skipped here?

File: Jenkinsfile.nightly [hunk 3]
```diff
@@ -44,5 +44,31 @@ pipeline {
...
         }
+
+        stage('Kubernetes tests') {
+            environment {
+                GOPATH="${WORKSPACE}"
+                TESTDIR="${WORKSPACE}/${PROJ_PATH}/test"
+            }
+            steps {
+                parallel(
+                    "K8s-1.7":{
```
Comment (ianvernon) [inline Jenkinsfile.nightly:55]: Liking the move of older K8s versions to nightly tests. What is the rationale behind keeping the 1.6 tests around as part of each PR? I'm for moving them to nightly as well.

Comment (eloycoto) [reply Jenkinsfile.nightly:55]: A lot of test has different cases for 1.6, the CNP v1, so I think that we should keep it.

File: test/provision/k8s_install.sh (hunk 1 above)
Comment (eloycoto) [reply test/provision/k8s_install.sh]: Is mandatory for kubernetes

Comment (ianvernon) [reply test/provision/k8s_install.sh]: What makes this mandatory? Please add a comment in the script to specify why this is necessary so future editors of the script have appropriate context.

File: Jenkinsfile.nightly (hunk 3 above)
Comment (ianvernon) [reply Jenkinsfile.nightly:55]: Fair enough.

**Comment by eloycoto** [comment]
test-all-ginkgo

**Comment by eloycoto** [comment]
test-all-ginkgo

**Comment by eloycoto** [comment]
test-all-ginkgo

**Comment by eloycoto** [comment]
test-me-please

**Comment by eloycoto** [comment]
test-all-ginkgo

**Comment by eloycoto** [comment]
test-me-please

**Review by aanm — APPROVED** [review]

**Pull request merged**